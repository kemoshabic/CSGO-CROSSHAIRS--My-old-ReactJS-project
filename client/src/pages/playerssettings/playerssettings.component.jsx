import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import Settings from "../../components/settings-overview/settings-overview.component";
import { selectFavoriteSettings } from "../../redux/settings/settings.selector";
import { removeSettings } from "../../redux/settings/settings.actions"

import PurpleHeart from "../../assets/purple.png";

import "./playerssettings.styles.scss";

export const PlayersSettings = ({ savedSettings, removeSettings }) => {
  return (
    <div className="settings">
      {savedSettings.map(({ team, name, id, settings }) => (
        <div className="settings__block" key={id}>
          <div className="settings__top">
            <img src={PurpleHeart} alt="hey" className="smh" />
            <h2 className="settings__heading">{name}</h2>
            <span className="settings__team">{team}</span>
            <p className="settings__delete" onClick={() => removeSettings({team, name, id, settings})}>Delete</p>
          </div>
          <Settings settings={settings}/>
        </div>
      ))}
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  savedSettings: selectFavoriteSettings,
});

const mapDispatchToProps = (dispatch) => ({
  removeSettings: settings => dispatch(removeSettings(settings)),
})


export default connect(mapStateToProps, mapDispatchToProps)(PlayersSettings);
