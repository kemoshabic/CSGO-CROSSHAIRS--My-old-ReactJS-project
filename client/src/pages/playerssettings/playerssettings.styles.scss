.settings{
  display: flex;
  flex-direction: column;
  &__block{
    padding: 5rem 11rem;
    border-bottom: 1px solid #FD074E;
    display: flex;
    flex-direction: column;
    @media only screen and (max-width: 103.12em) {
      // 1650px
      padding: 4rem 6rem;
    }
    @media only screen and (max-width: 57.5em) {
      // 920px
      padding: 3rem 4rem;
    }
  }
  &__top{
    display: flex;
    align-items: center;
    padding-bottom: 4rem;
    @media only screen and (max-width: 103.12em) {
      // 1650px
      padding-bottom: 2rem;
    }
  }
  &__heading{
    color: #FFFFFF;
    font-size: 2.8rem;
    padding-left: 2rem;
  }
  &__team{
    padding-left: 1.5rem;
    margin-top: 0.8rem;
    color: #FFFFFF;
    font-size: 1.8rem;
  }
  &__show{
    align-self: flex-start;
  }
  &__delete {
    font-size: 2rem;
    color: red;
    text-decoration: underline;
    padding-left: 1.8rem;
    margin-bottom: -0.5rem;
    cursor: pointer;
    &:hover {
      background-image: linear-gradient(to right, #80ff72, #7ee8fa);
      background-clip: text;
      color: transparent;
    }
  }
}
