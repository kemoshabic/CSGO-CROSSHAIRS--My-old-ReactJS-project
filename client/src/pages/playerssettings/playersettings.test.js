import React from "react";
import { shallow } from "enzyme";
import { PlayersSettings } from './playerssettings.component'
describe("PlayersSettings component", () => {
  let wrapper;
  let mockRemoveSettings;
  let mockSavedSettings = [
    {
      team: 'text',
      name: 'text',
      id: 1,
      settings: [],
    }
  ]
  beforeEach(() => {
    mockRemoveSettings = jest.fn(); 
    const mockProps={
      savedSettings: mockSavedSettings,
      removeSettings: mockRemoveSettings
    }
    wrapper = shallow(<PlayersSettings {...mockProps} />);
  });

  it('should render the PlayersSettings component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it("should call removeSettings function when button is clicked", () => {
    wrapper.find(".settings__delete").simulate("click");
    expect(mockRemoveSettings).toHaveBeenCalled();
  });
  
  it("should call removeSettings function when button is clicked", () => {
    wrapper.find(".settings__delete").simulate("click");
    expect(mockRemoveSettings).toHaveBeenCalled();
  });

});