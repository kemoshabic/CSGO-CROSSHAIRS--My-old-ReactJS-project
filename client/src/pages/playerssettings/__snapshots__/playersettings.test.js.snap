// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayersSettings component should render the PlayersSettings component 1`] = `ShallowWrapper {}`;
